hoot-core/src/main/cpp/hoot/core/algorithms/WayMatchStringSplitter.cpp

```diff
@@ -129,126 +113,15 @@ void WayMatchStringSplitter::_splitWay1(OsmMapPtr map, vector<pair<ElementId, El
     {
       assert(wls[i] <= wls[i + 1]);
       if (i % 2 == 1 && wls[i] != wls[i + 1])
-      {
         discontiguous = true;
-      }
     }
 
-    LOG_VART(wls);
-    vector<WayPtr> splits = WaySplitter(map, w1).createSplits(wls);
-
-    assert((int)splits.size() == sm.size() * 2 + 1);
-
-    QList<ElementPtr> newWays;
-
-    int c = 0;
-    WayPtr w = splits[c++];
-    // if this isn't empty
-    if (w && ec.calculateLength(w) > 0.0)
-    {
-      newWays.append(w);
-      replaced.push_back(pair<ElementId, ElementId>(w1->getElementId(), w->getElementId()));
-    }
-
-    for (int i = 0; i < sm.size(); ++i)
-    {
-      WayPtr w;
-      w = splits[c++];
-      if (!w)
-      {
-        throw NeedsReviewException(_overlyAggressiveMergeReviewText);
-      }
-      sm.at(i)->newWay1 = w;
-      replaced.push_back(pair<ElementId, ElementId>(w1->getElementId(), w->getElementId()));
-      newWays.append(w);
-
-      w = splits[c++];
-      // if this isn't empty
-      if (w && ec.calculateLength(w) > 0.0)
-      {
-        // only the last split should be non-empty
-        if (i != sm.size() - 1 && discontiguous == false)
-        {
-          throw InternalErrorException("Only the last split should be empty.");
-        }
-        newWays.append(w);
-        replaced.push_back(pair<ElementId, ElementId>(w1->getElementId(), w->getElementId()));
-      }
-    }
-
-    // see comments for similar functionality in HighwaySnapMerger::_mergePair
-    if (w1->getStatus() == Status::Unknown1 && _preserveUnknown1ElementIdWhenModifyingFeatures)
-    {
-      LOG_TRACE(
-        "Retaining reference ID by mapping unknown1 ID: " << w1->getElementId() << " to ID: " <<
-        newWays[0]->getElementId() << "...");
-      _unknown1Replacements.insert(
-        pair<ElementId, ElementId>(w1->getElementId(), newWays[0]->getElementId()));
-    }
-
-    LOG_TRACE("Replacing: " << w1->getElementId() << " with: " << newWays);
-    map->replace(w1, newWays);
-  }
-}
-
```

Comment:
The diff here might be a little confusing.  It thinks I deleted the bottom half of the first function and the top half of the bottom function.  Either way, there is only one copy and it is generic based on the `WayNumber` parameter.